src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -240,9 +240,46 @@ private static void SetMSBuildMetadata(IObjectWriter writer, PackageSpec package
                 writer.WriteObjectEnd();
             }
 
+            SetWarningProperties(writer, msbuildMetadata);
+
             writer.WriteObjectEnd();
         }
 
+        private static void SetWarningProperties(IObjectWriter writer, ProjectRestoreMetadata msbuildMetadata)
+        {
+            if (msbuildMetadata.ProjectWideWarningProperties != null &&
+                (msbuildMetadata.ProjectWideWarningProperties.AllWarningsAsErrors ||
+                 msbuildMetadata.ProjectWideWarningProperties.NoWarn.Count > 0 ||
+                 msbuildMetadata.ProjectWideWarningProperties.WarningsAsErrors.Count > 0))
+            {
+                writer.WriteObjectStart("warningProperties");
+
+                SetValueIfTrue(writer, "allWarningsAsErrors", msbuildMetadata.ProjectWideWarningProperties.AllWarningsAsErrors);
+
+                if (msbuildMetadata.ProjectWideWarningProperties.NoWarn.Count > 0)
+                {
+                    SetArrayValue(writer, "noWarn", msbuildMetadata
+                       .ProjectWideWarningProperties
+                       .NoWarn
+                       .ToArray()
+                       .OrderBy(c => c)
```

Comment:
👍 glad this is ordered here, I was worried about having to order on reading from the project